yarnpkg/yarn
# Do not try to resolve optionalDependencies defined with file:<path> when --ignore-optional is set

**Summary**
for an optional dependency using <file:PATH>, yarn --ignore-optional is failing returning the path does not exists. It should not:

https://github.com/yarnpkg/yarn/issues/2149



**Test plan**

- Ensure yarn --ignore-optional is not failing when an optionalDependency defined as __file:<path>__ path is not found
- Ensure yarn --ignore-optional is not installing optionalDependency defined as __file:<path>__ path even if it's resolved found
- Ensure yarn is failing when an dependency defined as __file:<path>__ path is not found
- Ensure yarn is installing a dependency defined as __file:<path>__ path when it's found
